ceph.spec.in

```diff
@@ -836,6 +833,20 @@ system.  It is responsible for storing objects on a local file system
 and providing access to them over the network.
 %endif
 
+%package volume
+Summary: Ceph OSD deployment and inspection tool
+BuildArch: noarch
+%if 0%{?suse_version}
+Group: System/Filesystems
+%endif
+Requires: lvm2
+Requires: parted
+Requires: python%{python3_pkgversion}-setuptools
```

Comment:
I think you forgot to add back the python-ceph-common dependency

```suggestion
Requires: python%{python3_pkgversion}-setuptools
Requires: python%{python3_pkgversion}-ceph-common = %{_epoch_prefix}%{version}-%{release}
```

The current change would allow one to :
- install ceph-volume automatically via ceph-osd package.
- install ceph-volume as a standalone package.

The second scenario will be broken (I'm not sure if that's a valid scenario) because we will have some missing dependencies like ceph-common (for ceph-authtool) or ceph-osd (for ceph-bluestore-tool) but I don't know if rpm allows to set circular dependencies.

Don't we need to update the ceph-osd scriptlet too ? Currently the pre/post tasks are using ceph-osd and ceph-volume systemd units.

I think we will need to update the ceph-volume requirements too (cryptsetup, util-linux,, etc..) but that's not the goal of that PR.

Reply:
@dsavineau i think i addressed all your comments

- ceph-volume depends on ceph-osd now instead the other way around
- ceph-osd does not depend on ceph-volume, but it recommends/suggests it. so `dnf` / `apt-get` can install the latter when installing the former.
- add cryptsetup, util-linux and xfsprogs as runtime dependencies of ceph-volume
- remove cryptsetup, util-linux and xfsprogs from the deps list of ceph-base.